util/file_reader_writer.cc

```diff
@@ -512,19 +513,24 @@ class ReadaheadRandomAccessFile : public RandomAccessFile {
         return Status::OK();
       }
     }
+    size_t aligned_offset = TruncateToPageBoundary(alignment_, offset + copied);
     Slice readahead_result;
-    Status s = file_->Read(offset + copied, readahead_size_, &readahead_result,
-      buffer_.get());
+    Status s = file_->Read(aligned_offset, readahead_size_ + alignment_,
+                           &readahead_result, buffer_.get());
```

Comment:
I don't understand this. Do we guarantee buffer_.get() is aligned? If this is Direct I/O mode, we need to keep it aligned to avoid this memory copy.